data/src/main/java/com/microsoft/azure/kusto/data/format/CslFormat.java

```diff
@@ -0,0 +1,22 @@
+package com.microsoft.azure.kusto.data.format;
+
+public abstract class CslFormat {

```

Comment:
I considered using generics here so we can have 1 `T value` property, but didn't since we need specialized behavior for the subclasses.